nodi_backend/apps/datasets/views.py

```diff
@@ -37,13 +40,23 @@ def retrieve(self, request, pk=None):
         dataset = get_object_or_404(queryset, pk=pk)
         serializer = DataSetSerializer(dataset)
         return JsonResponse(serializer.data)
-    
+
     def create(self, request):
         """
         +POST+
         Creates a dataset
         """
-        return HttpResponseNotFound("Not Implemented Yet")
+        body = json.loads(request.body)
+        d = DataSet(name=body['name'], email=body['email'],
+                    submitted_at=body['submitted_at'], approved=False, bucket='nodi-unapproved-datasets', key=body['key'], approved_at=None)
+        d.save()
+        d.add(Category.objects.get(id=body['category_ids'])
```

Comment:
So one thing that may be an issue here is that `get` expects a singular id. If you want multiple results you should use the `filter` function. This is actually really tricky to do so I'm gonna go in and fix it for you, because adding relations can cause a bunch of errors.